python/semantic_kernel/connectors/memory/redis/redis_memory_store.py

```diff
@@ -0,0 +1,530 @@
+
+    async def delete_all_collections_async(self, delete_records: bool = True) -> None:
+        """
+        Deletes all collections present in the data store.
+
+        Arguments:
+            delete_records {bool} -- Delete all data associated with the collections, default to True
+        """
+        for col_name in await self.get_collections_async():
+            self._ft(col_name).dropindex(delete_documents=delete_records)
+
+    async def does_collection_exist_async(self, collection_name: str) -> bool:
+        """
+        Determines if a collection exists in the data store.
+
+        Arguments:
+            collection_name {str} -- Name for a collection of embeddings
+
+        Returns:
+            True if the collection exists, False if not
+        """
+        try:
+            self._ft(collection_name).info()
+            return True
+        except ResponseError:
+            return False
+
+    async def upsert_async(self, collection_name: str, record: MemoryRecord) -> str:
+        """
+        Upsert a memory record into the data store. Does not gurantee that the collection exists.
+            * If the record already exists, it will be updated.
+            * If the record does not exist, it will be created.
+
+        Note: if the record do not have the same dimensionality configured for the collection,
+        it will not be detected to belong to the collection in Redis.
+
+        Arguemnts:
+            collection_name {str} -- Name for a collection of embeddings
+            record {MemoryRecord} -- Memory record to upsert
+
+        Returns
+            str -- Redis key associated with the upserted memory record, or None if an error occured
+        """
+        batch = await self.upsert_batch_async(collection_name, [record])
+        return batch[0] if len(batch) else None
+
+    async def upsert_batch_async(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        """
+        Upserts a group of memory records into the data store. Does not gurantee that the collection exists.
+            * If the record already exists, it will be updated.
+            * If the record does not exist, it will be created.
+
+        Note: if the records do not have the same dimensionality configured for the collection,
+        they will not be detected to belong to the collection in Redis.
+
+        Arguemnts:
+            collection_name {str} -- Name for a collection of embeddings
+            records {List[MemoryRecord]} -- List of memory records to upsert
+
+        Returns
+            List[str] -- Redis keys associated with the upserted memory records, or an empty list if an error occured
+        """
+
+        if not await self.does_collection_exist_async(collection_name):
+            self._logger.error(f'Collection "{collection_name}" does not exist')
+            raise Exception(f'Collection "{collection_name}" does not exist')
+
+        keys = list()
+        for rec in records:
```

Comment:
Since in `upsert_batch_async` method you use loop to upsert each record separately, wouldn't it be better to define this logic in `upsert_async` method for one record and then re-use it here within loop? Let me know if that makes sense.

Same applies for `get_async` and `get_batch_async` methods.